### PR TITLE
feat: support RN 0.75

### PR DESCRIPTION
**Pull request by WoLewicki**
## Description

PR adding support for RN 0.75 in the library

### CODE REVIEW

File: FabricExample/android/app/build.gradle [hunk 1]
```diff
@@ -74,9 +77,9 @@ android {
     buildToolsVersion rootProject.ext.buildToolsVersion
     compileSdk rootProject.ext.compileSdkVersion
 
-    namespace "com.fabricexample"
+    namespace "com.helloworld"
     defaultConfig {
-        applicationId "com.fabricexample"
+        applicationId "com.helloworld"
```
Comment (j-piasecki) [inline FabricExample/android/app/build.gradle:82]: That's what you get for making a new app every time 😃

File: FabricExample/ios/_xcode.env [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# This `.xcode.env` file is versioned and is used to source the environment
```
Comment (j-piasecki) [inline FabricExample/ios/_xcode.env:1]: That's not `.xcode.env`, it's `_xcode.env`

File: FabricExample/package.json [hunk 3]
```diff
@@ -19,8 +19,8 @@
     "@react-navigation/routers": "link:../react-navigation/packages/routers/",
     "@react-navigation/stack": "link:../react-navigation/packages/stack/",
     "nanoid": "^4.0.2",
-    "react": "18.2.0",
-    "react-native": "0.74.1",
+    "react": "19.0.0-rc-fb9a90fa48-20240614",
+    "react-native": "0.75.0-rc.0",
```
Comment (j-piasecki) [inline FabricExample/package.json:23]: Why rc.0?

File: FabricExample/ios/_xcode.env (hunk 2 above)
Comment (WoLewicki) [reply FabricExample/ios/_xcode.env:1]: I did not create it manually 🤷‍♂️

Comment (j-piasecki) [reply FabricExample/ios/_xcode.env:1]: But the other one exists and is basically the same thing

**Comment by raghav-reglobe** [comment]
Hi, when can we expect this to be merged.

**Comment by WoLewicki** [comment]
I'm not sure @raghav-reglobe ? What is the reason why you need it?

**Comment by raghav-reglobe** [comment]
@WoLewicki, thanks for asking.

I have migrated to react-native new architecture, but struggling with memory consumption, and animation performance(react-native-reanimated), Tried many things to optimize it.

So I am hoping for performance boost with react@19 updates, which is available in RN75, but unable to compile it, because of the few removed deprecated methods are being used this lib.

**Comment by WoLewicki** [comment]
OK I get it. I think the best way forward for now is to use [patch-package](https://www.npmjs.com/package/patch-package) to get the needed changes in your project.

**Comment by imWildCat** [comment]
Is there any plan to release the change of this PR as an rc/beta version for the community to test?

**Comment by WoLewicki** [comment]
We should release rc version in about 2 weeks. For now I suggest you do the thing mentioned in the [comment](https://github.com/software-mansion/react-native-screens/pull/2218#issuecomment-2223290658)

File: FabricExample/babel.config.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
   presets: ['module:@react-native/babel-preset'],
+  plugins: ['react-native-reanimated/plugin'],
```
Comment (kkafar) [inline FabricExample/babel.config.js:3]: Nice catch. I've created ticket for this which can now be closed.

File: android/src/fabric/java/com/swmansion/rnscreens/FabricEnabledViewGroup.kt [hunk 5]
```diff
@@ -48,14 +49,11 @@ abstract class FabricEnabledViewGroup constructor(
...
-            map.putDouble("contentOffsetX", 0.0)
-            map.putDouble("contentOffsetY", headerHeight)
-            map
-        }
+        val map: WritableMap = WritableNativeMap()
+        map.putDouble("frameWidth", realWidth.toDouble())
+        map.putDouble("frameHeight", realHeight.toDouble())
+        map.putDouble("contentOffsetX", 0.0)
+        map.putDouble("contentOffsetY", headerHeight)
+        mStateWrapper?.updateState(map)
```
Comment (kkafar) [inline android/src/fabric/java/com/swmansion/rnscreens/FabricEnabledViewGroup.kt:57]:
```suggestion
        val map: WritableMap = WritableNativeMap().apply {
            putDouble("frameWidth", realWidth.toDouble())
            putDouble("frameHeight", realHeight.toDouble())
            putDouble("contentOffsetX", 0.0)
            putDouble("contentOffsetY", headerHeight)
        }
        mStateWrapper?.updateState(map)
```

File: ios/RNSScreen.mm [hunk 6]
```diff
@@ -295,6 +295,14 @@ - (void)addSubview:(UIView *)view
   }
 }
 
+- (void)postNotificationWithEvent:(NSObject<RCTEvent> *)event
+{
+  NSDictionary *userInfo = [NSDictionary dictionaryWithObjectsAndKeys:event, @"event", nil];
+  [[NSNotificationCenter defaultCenter] postNotificationName:@"RCTNotifyEventDispatcherObserversOfEvent_DEPRECATED"
+                                                      object:nil
+                                                    userInfo:userInfo];
+}
```
Comment (kkafar) [inline ios/RNSScreen.mm:304]: I don't like naming of this method, as this method sends very specific notification & we should either have this reflected in method name (literally or with some other applicable adjective) or take this as a param. 

E.g. `postNotificationForEventDispatcherObserversWithEvent` or smth.

File: ios/RNSScreen.mm [hunk 7]
```diff
@@ -295,6 +295,14 @@ - (void)addSubview:(UIView *)view
   }
 }
 
+- (void)postNotificationWithEvent:(NSObject<RCTEvent> *)event
```
Comment (kkafar) [inline ios/RNSScreen.mm:298]: Also, this method could be placed in Fabric specific section & compiled only on Fabric.

**Review by kkafar — CHANGES_REQUESTED** [review]
Few remarks from looking at the code. 
I'll test runtime in a bit

File: ios/RNSScreen.mm (hunk 6 above)
Comment (kkafar) [reply ios/RNSScreen.mm:304]: Done

File: ios/RNSScreen.mm (hunk 7 above)
Comment (kkafar) [reply ios/RNSScreen.mm:298]: Done

File: android/src/fabric/java/com/swmansion/rnscreens/FabricEnabledViewGroup.kt (hunk 5 above)
Comment (kkafar) [reply android/src/fabric/java/com/swmansion/rnscreens/FabricEnabledViewGroup.kt:57]: Done

**Review by kkafar — APPROVED** [review]
Builded Paper/Fabric + iOS/Android combinations, did some sanity checks, seems ok! Thanks a lot!

**Comment by imWildCat** [comment]
awesome! thank you! 🫡

**Pull request merged**